src/transport/SessionDelegate.h

```diff
@@ -20,10 +20,26 @@
 
 namespace chip {
 
-class DLL_EXPORT SessionReleaseDelegate
+class DLL_EXPORT SessionDelegate
 {
 public:
-    virtual ~SessionReleaseDelegate() {}
+    virtual ~SessionDelegate() {}
+
+    enum class SessionUpdated : uint8_t
```

Comment:
`SessionUpdated::Migrate` and `SessionUpdated::DoNotMigrate` does not seem intuitive to me.

Usage intent seems to be the return code from OnSessionUpdate and based on the comment on that method, it says to be called when a new session to the same peer is established. I do not understand what 'Migration' means though and there is no usage in this PR that I can see.

Please re-work the naming a bit. Initial thoughts would be:
   - `OnSessionUpdated(...)` method name indicates "the session was updated". Instead, based on the description, this is "called when a new session to the same peer was created". So it should be named accordingly. Something like `DecideNewPeerSessionHandling` or `GetNewPeerSessionUsage` or simil;ar
   - once we have a new name, rename  SessionUpdate enum to something else including members to be be intuitive. If you only have yes/no, then the naming should be something like `enum class MigrateToNewSession` with `kYes/kNo` options.
   - I do not understand what 'Migrate' means. Needs explaining or a new word needs to be used (cannot suggest much because I do not understand it)